server/test/repository/ApplicationEventRepositoryTest.java

```diff
@@ -149,4 +149,33 @@ public void getEvents() {
     // NoteEvent wasn't set and should be available as an empty Optional.
     assertThat(gotEvent.getDetails().noteEvent()).isNotPresent();
   }
+
+  @Test
+  public void test_setNoteEvent() {
+    // Setup
+    Instant startInstant = Instant.now();
+    ProgramModel program = resourceCreator.insertActiveProgram("Program");
+    AccountModel actor = resourceCreator.insertAccount();
+    ApplicantModel applicant = resourceCreator.insertApplicantWithAccount();
+    ApplicationModel application = resourceCreator.insertActiveApplication(applicant, program);
+    application.setLatestNote("initial note");
+    application.save();
+
+    repo.setNote(application, ApplicationEventDetails.NoteEvent.create("some note"), actor);
+
+    // Execute
+    ImmutableList<ApplicationEventModel> gotEvents =
```

Comment:
**nitpick:** Thoughts on naming this applicationEvents?

Reply:
sure will change it